File: constants/uemoa-rates.ts
/**
 * Taux de référence UEMOA — configurables, jamais codés en dur dans les calculs.
 * Source : BCEAO / BRVM / données publiques 2025.
 * FR50 : ces valeurs sont les défauts affichés dans le simulateur.
 */
export const UEMOA_RATES = {
  // Épargne classique (comptes rémunérés, livrets)
  SAVINGS_ACCOUNT:     0.035,  // 3.5%
  // Bons du Trésor UEMOA (court terme ~3 mois)
  TREASURY_BILLS_3M:   0.055,  // 5.5%
  // Bons du Trésor UEMOA (moyen terme ~1 an)
  TREASURY_BILLS_1Y:   0.065,  // 6.5%
  // Obligations d'État UEMOA (long terme)
  GOVERNMENT_BONDS:    0.075,  // 7.5%
  // BRVM — rendement historique moyen
  BRVM_AVERAGE:        0.085,  // 8.5%
  // Inflation UEMOA (référence BCEAO)
  INFLATION:           0.030,  // 3.0%
  // Taux de crédit à la consommation (référence remboursement dettes)
  CONSUMER_CREDIT:     0.150,  // 15%
} as const;

export type UEMOARateKey = keyof typeof UEMOA_RATES;

File: lib/core/constants/uemoa_rates.dart
/// Taux de référence UEMOA — FR50 : configurables, jamais codés en dur
/// dans les calculs du moteur de simulation.
///
/// Sources : BCEAO, BRVM, données publiques.
/// Ces valeurs alimentent les champs du simulateur par défaut.
abstract final class UemoaRates {
  /// Épargne classique — comptes rémunérés et livrets.
  static const savingsAccount = 0.035;

  /// Bons du Trésor UEMOA à court terme (≈ 3 mois).
  static const treasuryBills3m = 0.055;

  /// Bons du Trésor UEMOA à moyen terme (≈ 1 an).
  static const treasuryBills1y = 0.065;

  /// Obligations d'État UEMOA à long terme.
  static const governmentBonds = 0.075;

  /// BRVM — rendement historique moyen.
  static const brvmAverage = 0.085;

  /// Inflation de référence BCEAO.
  static const inflation = 0.030;

  /// Crédit à la consommation — référence pour les plans de remboursement.
  static const consumerCredit = 0.150;

  /// Libellés affichables des placements proposés dans le simulateur.
  static const Map<String, ({String label, double rate})> instruments = {
    'savings': (label: 'Épargne classique', rate: savingsAccount),
    'treasury_3m': (label: 'Bons du Trésor 3 mois', rate: treasuryBills3m),
    'treasury_1y': (label: 'Bons du Trésor 1 an', rate: treasuryBills1y),
    'bonds': (label: "Obligations d'État", rate: governmentBonds),
    'brvm': (label: 'Actions BRVM', rate: brvmAverage),
  };
}
